index.js

```diff
@@ -438,17 +438,20 @@ class Body extends events.EventEmitter {
     }
     this.ct = ct;
 
+    const buildCTRegex = (key) => new RegExp(`${key}\\s*=\\s*"([^"]+)"|${key}\\s*=\\s*"?([^";]+)"?`, 'i');
+    const matchCTKey = (test, key) => test.match(buildCTRegex(key))?.filter(item => item);
+
     let match;
     if (/^(?:text|message)\//i.test(ct) && !/^attachment/i.test(cd)) {
       this.state = 'body';
     } else if (/^multipart\//i.test(ct)) {
-      match = ct.match(/boundary\s*=\s*"?([^";]+)"?/i);
+      match = matchCTKey(ct, 'boundary');
       this.boundary = match ? match[1] : '';
       this.state = 'multipart_preamble';
     } else {
-      match = cd.match(/name\s*=\s*"?([^";]+)"?/i);
```

Comment:
This RE is testing against `cd`, but in this PR it's testing against `ct`. Looks like an error to me.

Perhaps your RE should be named `matchKey` instead, so it doesn't give false impressions?

Reply:
It is a mistake indeed, thank you @msimerson! Fixed.

Reply:
What do you think of renaming buildCTRegex and matchCTkey, since they're used to match against `ct` and `cd`?

Reply:
I thought the same when fixing it, but couldn't find a good name right off the bat (thought of `matchCTDKey`...). Any suggestion?

Reply:
Sometimes, simplest is best. How about `buildRegex` and `matchKey`?  (Yeah, the names are vague, but they're also encapsulated in their context, so I think that's okay.)

Reply:
Works for me, that was actually my first naming before adding CT. :) Done.